File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/text/TextStyle.java
package pize.tests.voxelgame.main.text;

public class TextStyle{
    
    private byte style;
    
    public TextStyle(){
        style = 0;
    }
    
    public TextStyle(TextStyle textStyle){
        style = textStyle.style;
    }
    
    public TextStyle(byte data){
        style = data;
    }
    
    
    public void enable(StyleFormatting style){
        this.style |= (1 << style.styleID);
    }
    
    public void reset(){
        style = 0;
    }
    
    
    public boolean isItalic(){
        return (style & (1 << StyleFormatting.ITALIC.styleID)) == 1;
    }
    
    public boolean isBold(){
        return (style & (1 << StyleFormatting.BOLD.styleID)) == 1;
    }
    
    
    public TextStyle copy(){
        return new TextStyle(this);
    }
    
    public byte getData(){
        return style;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/text/TextStyle.kt
package pize.tests.voxelgame.main.text

class TextStyle {
    var data: Byte
        private set

    constructor() {
        data = 0
    }

    constructor(textStyle: TextStyle) {
        data = textStyle.data
    }

    constructor(data: Byte) {
        this.data = data
    }

    fun enable(style: StyleFormatting) {
        data = (data.toInt() or (1 shl style.styleID)).toByte()
    }

    fun reset() {
        data = 0
    }

    val isItalic: Boolean
        get() = data.toInt() and (1 shl StyleFormatting.ITALIC.styleID) == 1
    val isBold: Boolean
        get() = data.toInt() and (1 shl StyleFormatting.BOLD.styleID) == 1

    fun copy(): TextStyle {
        return TextStyle(this)
    }
}
